Remotion/Data/DomainObjects/Infrastructure/DomainObjectTransactionContextExecutor.cs

```diff
@@ -0,0 +1,152 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Remotion.Data.DomainObjects.DataManagement;
+using Remotion.Data.DomainObjects.Persistence;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Infrastructure
+{
+  /// <summary>
+  /// Represents the transaction execution context of a <see cref="DomainObject"/> that is associated with a specific <see cref="ClientTransaction"/>.
+  /// </summary>
+  public readonly struct DomainObjectTransactionContextExecutor
```

Comment:
Executor does not fit well together with design IMO. We need to find a better name or even revert back to `DomainObjectTransactionContext`. Another todo that we should talk with mike about.